modules/core/module.go

```diff
@@ -35,7 +35,17 @@ var (
 )
 
 // AppModuleBasic defines the basic application module used by the ibc module.
-type AppModuleBasic struct{}
+type AppModuleBasic struct {
+	additionalInterfaceRegistrationFns []func(registry codectypes.InterfaceRegistry)
+}
+
+// NewAppModuleBasic creates a new AppModuleBasic object.
+// all functions provided will be executed in AppModuleBasic.RegisterInterfaces.
+func NewAppModuleBasic(additionalInterfaceRegistrationFns []func(codectypes.InterfaceRegistry)) AppModuleBasic {
```

Comment:
Does it make sense to use the `...` here?
```suggestion
func NewAppModuleBasic(additionalInterfaceRegistrationFns func(codectypes.InterfaceRegistry)...) AppModuleBasic {
```